test/e2e/README.md

```diff
@@ -2,45 +2,74 @@
+1. Each E2E test suite installs an instance of Velero to run tests and uninstalls it after test completion. It is possible that a test suite may be installing Velero while another may be uninstalling Velero. This race condition can lead to tests being flaky and cause false negatives. The options for resolving this are:
+    1. Make each test suite setup wait for Velero to be uninstalled before attempting to install as part of its setup.
+    1. Make each test suite install Velero in a different namespace.
 
-## Running tests based on cloud platforms
+## Configuration for E2E tests
 
-1. Running Velero E2E tests on KinD
+Below is a list of the configuration used by E2E tests.
+These configuration parameters are expected as values to the following command line flags:
 
-    ```
-    CLOUD_PLATFORM=kind make test-e2e
-    ```
+1. `-credentials-file`: File containing credentials for backup and volume provider. Required.
+1. `-bucket`: Name of the object storage bucket where backups from e2e tests should be stored. Required.
+1. `-plugin-provider`: Provider of object store and volume snapshotter plugins. Required.
+1. `-velerocli`: Path to the velero application to use. Optional, by default uses `velero` in the `$PATH`
+1. `-velero-image`: Image for the velero server to be tested. Optional, by default uses `velero/velero:main`
+1. `-bsl-config`: Configuration to use for the backup storage location. Format is key1=value1,key2=value2. Optional.
+1. `-prefix`: Prefix in the `bucket`, under which all Velero data should be stored within the bucket. Optional.
+1. `-vsl-config`: Configuration to use for the volume snapshot location. Format is key1=value1,key2=value2. Optional.
 
-1. Running Velero E2E tests on AWS
+These configurations or parameters are used to generate install options for Velero for each test suite.
 
-    ```
-    CLOUD_PLATFORM=aws make test-e2e
-    ```
+## Running tests locally
+
+### Running using `make`
+
+E2E tests can be run from the Velero repository root by running `make test-e2e`. While running E2E tests using `make` the E2E test configuration values are passed using `make` variables.
 
-1. Running Velero E2E tests on Azure
+Below is a mapping between `make` variables to E2E configuration flags.
+1. `CREDS_FILE`: `-credentials-file`. Required.
+1. `BSL_BUCKET`: `-bucket`. Required.
+1. `PLUGIN_PROVIDER`: `-plugin-provider`. Required.
+1. `VELERO_CLI`: the `-velerocli`. Optional.
+1. `VELERO_IMAGE`: the `-velero-image`. Optional.
+1. `BSL_PREFIX`: `-prefix`. Optional.
+1. `BSL_CONFIG`: `-bsl-config`. Optional.
+1. `VSL_CONFIG`: `-vsl-config`. Optional.
 
+For example, E2E tests can be run from Velero repository roots using the below command:
+
+1. Run Velero tests using AWS as the storage provider:
+    ```bash
+    BSL_PREFIX=<PREFIX_UNDER_BUCKET> BSL_BUCKET=<BUCKET_FOR_E2E_TEST_BACKUP> CREDS_FILE=/path/to/aws-creds GINKGO_FOCUS="AWS" make test-e2e
```

Comment:
Given that the provider specific tests have been changed into a generic test where the provider is specified, the `GINKGO_FOCUS` flag should be removed and changed to `PLUGIN_PROVIDER`. Same on line 67 for the Azure instructions.

Reply:
Yes, updated.